src/Service/SecurityAdvisoryWorker.php

```diff
@@ -35,6 +36,14 @@ public function __construct(Locker $locker, LoggerInterface $logger, ManagerRegi
     public function process(Job $job, SignalHandler $signal): array
     {
         $sourceName = $job->getPayload()['source'];
+        $package = $this->doctrine->getRepository(Package::class)->findOneBy(['id' => $job->getPackageId()]);
+
+        if (!$package instanceof Package) {
+            $this->logger->info('Security advisory update failed, skipping', ['packageId' => $job->getPackageId()]);
+
+            return ['status' => Job::STATUS_ERRORED, 'message' => 'Security advisory update failed, skipped'];
+        }
+
         $lockAcquired = $this->locker->lockSecurityAdvisory($sourceName);
```

Comment:
Again, didn't really have something like GitHub in mind when building this. This currently only allows a single GitHub package to be processed at once. This will probably lead to a backlog of jobs. This should probably differ whether the source handles multiple package or just a single package and get a different lock key.